### PR TITLE
Add support for Sidekiq 4

### PR DESCRIPTION
**Pull request by dsander**
The specs will only pass on sidekiq 4, not sure how and if we need to also keep support for sidekiq 3.

### CODE REVIEW

**Comment by bigzed** [comment]
I don't think we have to support Sidekiq 3 in the master branch and if someone wants to fix something for Sidekiq 3 he/she can still branch of from the last Sidekiq 3 version.

**Comment by bigzed** [comment]
Btw, do we wanna move forward on this and finally release it :laughing:

**Comment by dsander** [comment]
@bigzed Probably a good idea, I'll remove sidekiq 3 support and we will release a new minor version or even `1.0.0`?

**Comment by bigzed** [comment]
@dsander Actually, lets add support for sidekiq5 first and then release it.

File: lib/sidekiq/repeat/repeatable.rb [hunk 1]
```diff
@@ -3,17 +3,24 @@ module Repeat
...
       module ClassMethods
         def repeat(&block)
-          @cronline = MiniIceCube::MainDsl.new.instance_eval(&block).to_s
+          @block = block
+        end
+
+        def cronline
+          return @cronline if @cronline
+          return if @block.nil?
+          @cronline = MiniIceCube::MainDsl.new.instance_eval(&@block).to_s
```
Comment (maltoe) [inline lib/sidekiq/repeat/repeatable.rb]: why did this need this to be delayed?

Comment (dsander) [inline lib/sidekiq/repeat/repeatable.rb]: I think the main reason was to allow accessing the database inside the `repeat` block which does not have to be initialized when the file is required.

**Comment by bigzed** [comment]
Looks good to me, I merge it.

**Pull request merged**